pkg/k8s/k8s.go

```diff
@@ -561,33 +607,41 @@ func (c *cluster) collectComponents(ctx context.Context, labels map[string]strin
 	return components, nil
 }
 
-func PodInfo(pod corev1.Pod, labelSelector string) (*bom.Component, error) {
-	containers := make([]bom.Container, 0)
-	for _, s := range pod.Status.ContainerStatuses {
-		imageName, err := utils.ParseReference(s.Image)
-		if err != nil {
-			slog.Warn(fmt.Sprintf("unable to parse image reference, skipping: %s", s.Image))
-			continue
-		}
-		imageID := getImageID(s.ImageID, s.Image)
-		if len(imageID) == 0 {
-			continue
-		}
-		imageRef, err := utils.ParseReference(imageID)
-		if err != nil {
-			slog.Warn(fmt.Sprintf("unable to parse image reference, skipping: %s", s.Image))
-			continue
-		}
-		hex := imageRef.Context().Digest(imageRef.Identifier()).DigestStr()
-		// skip non sha256 digests
-		if len(hex) != digest.Canonical.Size()*2 {
+func getImageIDsByStatuses(pod corev1.Pod) []string {
+	ids := make([]string, len(pod.Spec.Containers))
+	if len(pod.Spec.Containers) == 1 && len(pod.Status.ContainerStatuses) == 1 {
+		ids[0] = getImageID(pod.Status.ContainerStatuses[0].ImageID)
+		return ids
+	}
+
+	statusMap := make(map[string]string)
+	for _, status := range pod.Status.ContainerStatuses {
+		statusMap[status.Name] = status.ImageID
+	}
+
+	for i, container := range pod.Spec.Containers {
+		if id, ok := statusMap[container.Name]; ok {
+			ids[i] = getImageID(id)
 			continue
 		}
-		co, err := GetContainer(hex, imageName)
+		ids[i] = getImageID(container.Image)
+	}
+
+	return ids
```

Comment:
We can add tests for these in `TestPodInfo`.

Reply:
added some test cases https://github.com/aquasecurity/trivy-kubernetes/pull/452/commits/028d56ae5600e349d1736ebc6328e2b7ca4bf123 https://github.com/aquasecurity/trivy-kubernetes/pull/452/commits/a392a06b815c182bae96b26377b28b123f372c1e